### PR TITLE
[MSI] Add power state after power fail EFI variable

### PR DESCRIPTION
**Pull request by SergiiDmytruk**
I guess this will need EDK2 version bump when https://github.com/Dasharo/DasharoModulePkg/pull/25 is merged.

### CODE REVIEW

**Comment by miczyg1** [comment]
> I guess this will need EDK2 version bump when [Dasharo/DasharoModulePkg#25](https://github.com/Dasharo/DasharoModulePkg/pull/25) is merged.

You should have done that regardless of https://github.com/Dasharo/DasharoModulePkg/pull/25 state @SergiiDmytruk

File: src/superio/nuvoton/nct6687d/superio.c [hunk 1]
```diff
@@ -21,6 +23,34 @@
 #define NCT6687D_ACPI_POWER_PREV_STATE		(2 << NCT6687D_ACPI_POWER_LOSS_CONTROL_SHIFT)
 #define NCT6687D_ACPI_POWER_USER_DEFINED	(3 << NCT6687D_ACPI_POWER_LOSS_CONTROL_SHIFT)
 
+static uint8_t get_power_on_after_fail(void)
```
Comment (miczyg1) [inline src/superio/nuvoton/nct6687d/superio.c:26]: pmc_set_power_failure_state may also need this

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: src/superio/nuvoton/nct6687d/superio.c (hunk 1 above)
Comment (SergiiDmytruk) [reply src/superio/nuvoton/nct6687d/superio.c:26]: Hm, any suggestion where to put the implementation?  These EFI variables seem to be all over the place, so maybe the rest/new ones can be put there too eventually.  Maybe to something like `src/lib/dasharo.c`?

Comment (miczyg1) [reply src/superio/nuvoton/nct6687d/superio.c:26]: I was thinking about putting the implementation in one place too. I think more appropriate place would be the `src/vendorcode/dasharo/options.c` or something like that.

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: src/vendorcode/dasharo/options.c [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * a default before the call. */
+static enum cb_err read_u8_var(const char *var_name, uint8_t *var)
+{
+	struct region_device rdev;
+
+	if (!smmstore_lookup_region(&rdev)) {
+		uint8_t tmp;
+		uint32_t size = sizeof(tmp);
+		enum cb_err ret = efi_fv_get_option(&rdev, &dasharo_system_features_guid,
+						    var_name, &tmp, &size);
```
[the remark this answers is not in the document]
Comment (SergiiDmytruk) [reply src/vendorcode/dasharo/options.c:24]: Added the check. https://github.com/Dasharo/coreboot/compare/e95a507ef6393071918e1801bd58428a631f16c5..2be06e018f659e608936712a565373cc3baeaf90#diff-9219efc7069e199d2a4ae6d648d4f9986874aaed38374fa1f2567f9971d5a305R26

**Review by miczyg1 — APPROVED** [review]

**Pull request merged**